packages/react/src/components/DashboardEditor/DashboardEditorCardRenderer.test.jsx

```diff
@@ -231,40 +234,44 @@ describe('DashboardEditorCardRenderer', () => {
     expect(screen.getByText(/defaultCard/)).toBeInTheDocument();
   });
 
-  it('should call onFetchDynamicDemoHotspots when the function is available and dynamic hotspots are passed', () => {
+  it('should call onFetchDynamicDemoHotspots when the function is available and dynamic hotspots are passed', async () => {
     const onCardChange = jest.fn();
 
     const onFetchDynamicDemoHotspots = jest
       .fn()
       .mockImplementation(
-        () => new Promise((resolve) => resolve([{ x: 75, y: 10, type: 'text' }]))
+        () =>
+          new Promise((resolve) =>
+            resolve([{ x: 75, y: 10, type: 'text', content: { title: 'Title' } }])
+          )
       );
-
-    render(
-      <DashboardEditorCardRenderer
-        {...commonProps}
-        size={CARD_SIZES.MEDIUM}
-        type="IMAGE"
-        onCardChange={onCardChange}
-        content={{
-          src: 'landscape',
-          image: 'landscape',
-          alt: 'Sample image',
-          zoomMax: 10,
-          hasInsertFromUrl: true,
-        }}
-        onFetchDynamicDemoHotspots={onFetchDynamicDemoHotspots}
-        values={{
-          hotspots: [
-            {
-              color: 'purple',
-              icon: 'Checkmark',
-              type: 'dynamic',
-            },
-          ],
-        }}
-      />
-    );
+    await act(async () => {
+      render(
+        <DashboardEditorCardRenderer
+          {...commonProps}
+          size={CARD_SIZES.MEDIUM}
+          type="IMAGE"
+          onCardChange={onCardChange}
+          content={{
+            src: 'landscape',
+            image: 'landscape',
+            alt: 'Sample image',
+            zoomMax: 10,
+            hasInsertFromUrl: true,
+          }}
+          onFetchDynamicDemoHotspots={onFetchDynamicDemoHotspots}
+          values={{
+            hotspots: [
+              {
+                color: 'purple',
+                icon: 'Checkmark',
+                type: 'dynamic',
+              },
+            ],
+          }}
+        />
+      );
+    });
```

Comment:
wrapped everything in act to handle the promise callback being called on render and causing a state-change that wasn't wrapped in act error.